Let Uncrustify format C++-style comments

stinos opened the pull request:
Enables adding space after `//`.
I first got rid of all affected dead code for printf/assert/..., then formatted.
Some of the changes are still in what's quite dead code but sometimes it's also hard to tell if it can serve as a comment or not so I just left that in.

dlech commented on extmod/nimble/modbluetooth_nimble.c:
```diff
@@ -44,7 +44,7 @@
 #define MICROPY_PY_BLUETOOTH_DEFAULT_NAME "PYBD"
 #endif
 
-#define DEBUG_EVENT_printf(...) //printf(__VA_ARGS__)
```
I don't think this is dead code, I think it is for enabling debug prints.

A common pattern I have seen is something like this:

```C
#define DEBUG_EVENT 0
#if DEBUG_EVENT
#define DEBUG_EVENT_printf(...) printf(__VA_ARGS__)
#else
#define DEBUG_EVENT_printf(...)
#endif
```

They you just have to change 0 to 1 to enable debug printing. Maybe this would be better?

stinos replied:
Yes that's exactly what I'm saying in the commit message. But same remark: who knows which ones are really worth it?

dlech reviewed (commented):
I'm not sure about removing all of the debug prints. I occasionally find them useful to know what are the important values here when there is a problem.

stinos commented:
I get that point, but for every one that might be useful there seems to be another one which was just left there with no use. I admit I couldn't be bothered to go and check the full git log of all of them to figure out what it could be :] Since I mainly wanted to get formatting to work here either all can stay or none and that is my favorite so I chose that.. But if it's a true hindrance they can all just be formatted of course, although that's a bit silly in the case of those which are not ever going to be used (and it's not like the others cannot be found and put back if desired).

tve commented on ports/stm32/modnwwiznet5k.c:
```diff
@@ -191,11 +191,8 @@ STATIC int wiznet5k_socket_accept(mod_network_socket_obj_t *socket, mod_network_
             } else if (wiznet5k_socket_bind(socket, NULL, *port, &_errno2) != 0) {
-                //printf("(bad rebind %d)\n", _errno2);
             } else if (wiznet5k_socket_listen(socket, 0, &_errno2) != 0) {
-                //printf("(bad relisten %d)\n", _errno2);
```
Are you planning to leave the empty if statements?

stinos replied:
I guess, what would ba another option?

tve replied:
removing them...

stinos replied:
I'm hesitant to do that since above it states `TODO handle errors` so I assume at one point or another these empty statements are going to get filled with `return -1` or so?

jimmo reviewed (commented):
I am in favour of this change but would prefer not to remove any of the commented out code (e.g. the printf's)

dpgeorge commented on ports/esp8266/esppwm.c:
```diff
@@ -20,7 +20,6 @@
 
 #include "py/mpprint.h"
 #define PWM_DBG(...)
-//#define PWM_DBG(...) mp_printf(&mp_plat_print, __VA_ARGS__)
```
As above, probably better to add `#if 0 ... ` here.

dpgeorge commented on ports/esp8266/ets_alt_task.c:
```diff
@@ -142,34 +139,27 @@ bool ets_loop_iter(void) {
         if (t->i_get != t->i_put) {
             progress = true;
-            //printf("#%d Calling task %d(%p) (%x, %x)\n", cnt++,
-            //    t - emu_tasks + FIRST_PRIO, t->task, t->queue[t->i_get].sig, t->queue[t->i_get].par);
```
This one should be converted to `DEBUG_printf` style, but that's a bit more involved here.... so probably best for a separate commit (which I could do).

dpgeorge commented on ports/unix/main.c:
```diff
@@ -548,13 +548,6 @@ MP_NOINLINE int main_(int argc, char **argv) {
-    printf("    total %d\n", m_get_total_bytes_allocated());
-    printf("    cur   %d\n", m_get_current_bytes_allocated());
-    printf("    peak  %d\n", m_get_peak_bytes_allocated());
-    */
```
Does this need to be removed?  Is it auto-formatted?

dpgeorge commented on tools/uncrustify.cfg:
```diff
@@ -853,7 +853,7 @@ sp_extern_paren                 = ignore   # ignore/add/remove/force
 # Add or remove space after the opening of a C++ comment,
 # i.e. '// A' vs. '//A'.
-sp_cmt_cpp_start                = ignore   # ignore/add/remove/force
+sp_cmt_cpp_start                = add   # ignore/add/remove/force
```
This should probably be "force" which means "make sure there is only one".

stinos replied:
True, that's an oversight of mine, I think I was testing if it made a difference and than forgot to put it back.

dpgeorge commented:
I do think this is a good change/improvement, but it does mean that `//` should now (moving forward) only be used for comments, and not for commented-out code.  But IMO that's good because 1) it means the code will be cleaner because there are no commented-out bits, and no wondering why it was commented out; 2) it forces a more consistent and formal way of adding debug code, and code that is unused for some reason.

But to get there, I don't think all existing commented-out code should be deleted.  Instead it can be reformatted to use `DEBUG_printf`, for example.  That is a bit more effort though, and no expectation for it to be done here.

stinos commented:
Gave this some more thought in the meantime and I agree that just stripping what looks useless (to me) is not ideal here. I'll adress the straightforward fixes and then see what's left, I guess it's no huge problem if some of the commented out code then stays and eventually gets fixed later on.

stinos commented:
@dpgeorge what with the 'special' formatted comments like in bc.h, bc0.h, obj.h?

dpgeorge commented:
> what with the 'special' formatted comments like in bc.h, bc0.h, obj.h?

Hmm, ok, so maybe the uncrustify option needs to be "add" rather than "force".

stinos commented:
That does reduce the diff, and is perhaps better overall since it make it possibly to apply some structure to comments and breaks less things. Other problematic cases: wherever Python code is shown in the comments it ruins the indenting of it, there's some places where the slash is actually used for division i.e. we have
```
// Python sample:
// class Foo:
//     pass
// Operations:
// + is a sum
// / is divide
```
which would be turned into
```
// Python sample:
// class Foo:
// pass
// Operations:
// + is a sum
/// is divide
```

stinos committed:
all: Add space after C++-style comment start.

Note the Uncrustify configuration is explicitly set to 'add'
instead of 'force' in order not to alter the comments which use
extra spaces after // as a means of indenting text for clarity.

stinos committed:
all: Remove commented out include statements.

stinos commented:
Redid this: all assert/printf just stay in now. From the comments above there were only 2 left in code which actually got changed by formatting so I left that as is. Only left the removal of commented out include files because that really doesn't serve a purpose the compiler cannot do.

dpgeorge commented:
This looks OK now, I'd be happy to merge it.

tve commented:
What happened? Did this get merged?

stinos commented:
@tve Looks like I accidentally deleted the remote branch instead of my local one :)
But now Github doesn't allow me to reopen the PR, is that normal?

dlech commented:
I did that before too. I don't remember exactly what I did to fix it. Maybe push a new remote branch with the same name?

tve commented:
UNless you have changes to push, I believe you can leave it as-is and DP can still merge it.

dpgeorge commented:
I can't reopen it but the branch is still there to pull, so I can pull it in.

dpgeorge commented:
Merged in d6243568a05d423b58522435c3779975acbf56dd and 84fa3312cfa7d2237d4b56952f2cd6e3591210c4

stinos commented:
Thanks!

> I can't reopen it

So this is actually a bug then? I assumed it was a policy you set on the repo..

dpgeorge commented:
> I assumed it was a policy you set on the repo..

No, not a policy that I know of...
